akka-actor/src/main/scala/akka/pattern/Patterns.scala

```diff
@@ -525,4 +536,14 @@ object PatternsCS {
    */
   def after[T](duration: FiniteDuration, scheduler: Scheduler, context: ExecutionContext, value: CompletionStage[T]): CompletionStage[T] =
     afterCompletionStage(duration, scheduler)(value)(context)
+
+  /**
+   * Returns an internally retrying [[java.util.concurrent.CompletionStage]]
+   * The first attempt will be made immediately, each subsequent attempt will be made after 'delay'
+   * A scheduler (eg context.system.scheduler) must be provided to delay each retry
+   * If attempts are exhausted the returned completion stage is simply the result of invoking attempt
+   */
+  def retry[T](attempt: CompletionStage[T], attempts: Int, delay: FiniteDuration, scheduler: Scheduler, ec: ExecutionContext): CompletionStage[T] =
+    scalaRetry(() ⇒ attempt.toScala, attempts, delay, scheduler)(ec).toJava
```

Comment:
isn't this incorrect, since we keep returning the same future, but rather we should be creating a new attempt?
The attempt type should likely be `Callable[...` as above right?